pkg/kube/interface.go

```diff
@@ -54,7 +54,7 @@ type Interface interface {
 
 	// Update updates one or more resources or creates the resource
 	// if it doesn't exist.
-	Update(original, target ResourceList, force bool) (*Result, error)
+	Update(original, target ResourceList, force bool, threeWayMergeForUnstructured bool) (*Result, error)
```

Comment:
Since we are targeting to fix this in v3, this API change will break backward compatibility

Reply:
Would like me to add a method to the interface and use that new method instead of adapting the existing method to preserve backward compatibility?

Reply:
I'll bring this PR up in the developer call tomorrow. Depending on the outcome, you might not need to make any change to this function.

Notes: https://docs.google.com/document/d/1d-6xJEx0C78csIYSPKJzRPeWaHG_8W1Hjl72OJggwdc/edit?tab=t.0#heading=h.cafxi2yfgqg1

If we can default to 3-way merging, the additional `threeWayMergeForUnstructured` parameter may not be needed. If we need the parameter, the way to extend the API is to create a new interface which kube Client implements. Take a look at [here](https://github.com/helm/helm/blob/dev-v3/pkg/kube/interface.go#L78) and [here](https://github.com/helm/helm/blob/dev-v3/pkg/kube/interface.go#L125). In golang, extending an interface is a breaking change because any type implementing that interface will break

Reply:
@p-se remember to add `// TODO Helm 4: Remove this...` in the new interface, similar to the other extensions

Reply:
@p-se from today's developer call, it was agreed not to have to 3-way merging for CRDs and unstructured resources as the default behaviour so as not to surprise users with a change in behaviour.

We will keep the `threeWayMergeForUnstructured` boolean that will be `true` whenever `--take-ownership` flag is set.

Reply:
Please create a new interface like below (or similar) that `kube.Client` implements, then revert the previous change you made. Once that's done this PR should be good to go

```go
// TODO Helm 4: Remove InterfaceThreeWayMerge and integrate its method(s) into the Interface.
type InterfaceThreeWayMerge interface {
	UpdateThreeWayMerge(original, target ResourceList, force bool, threeWayMergeForUnstructured bool) (*Result, error)
}
```